app/models/foreman_tasks/triggering.rb

```diff
@@ -39,17 +35,27 @@ class Triggering < ActiveRecord::Base
                         :message => _("%{value} is wrong format"), :allow_blank => true
     validates_format_of :days, :with => DAYS_REGEXP,
                         :if => Proc.new { |t| t.recurring? && t.input_type == :monthly }
-    validate :correct_cronline, :if => Proc.new { |t| t.recurring? && t.input_type == :cronline }
+    validate :can_start_recurring, :if => :recurring?
+    validate :can_start_future, :if => :future?
 
     def self.new_from_params(params = {})
       self.new(params.except(:mode, :start_at, :start_before)).tap do |triggering|
         triggering.mode = params.fetch(:mode, :immediate).to_sym
         triggering.input_type = params.fetch(:input_type, :daily).to_sym
         triggering.end_time_limited = params[:end_time_limited] == "true"
         triggering.start_at_raw ||= Time.now.strftime(TIME_FORMAT)
+        triggering.recurring_logic = ::ForemanTasks::RecurringLogic.new_from_triggering(triggering) if triggering.recurring?
       end
     end
 
+    def mode
+      read_attribute(:mode).to_sym
```

Comment:
`super.to_sym` might work

also you could use your own serializer (only if you prefer, I don't have strong opinion) - http://apidock.com/rails/ActiveRecord/AttributeMethods/Serialization/ClassMethods/serialize#1343-Custom-serialization